gateway-ha/src/main/java/io/trino/gateway/baseapp/WebUIStaticResource.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.baseapp;
+
+import jakarta.servlet.ServletContext;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static jakarta.ws.rs.core.Response.Status.NOT_FOUND;
+
+@Path("/")
```

Comment:
this will prevent forwarding `/` to a backend correct? In some cases I forward `/`, and access the TGW UI on the app port. IMO we should put all the UI assets behind a prefix such as `/tgw`, and make the prefix configurable if possible

Reply:
That's not an easy task. IIUC, the react webapp hard coded `/` as root path in many places. Airlift doesn't support setting the ContextPath for ServletContextHandler, which makes configuring the prefix difficult.

IMO forwarding `/`, `/v1/info`, etc to an arbitrary Trino cluster is a bit confusing. Something like `/route/{trinoClusterName}/ui` might make more sense, except this break the drop-in transparent proxy function. Could you explain more about the use case?

Reply:
Starburst has a web UI with multi-cluster query history support, which also includes a SQL IDE. I agree that it is not useful to forward to an arbitrary OS Trino WebUI. I forward `/` because this is redirected to the webUI, its more user friendly for a user to type into their browser than remembering the path.

This may also apply to products such as DuneSQL, which I'm less familiar with.